### PR TITLE
Added the Entry.IsTextPredictionEnabled property

### LINKED ISSUE #1677: [Enhancement] Entry: Control over text-prediction
## Rationale

`Entry`s do not currently allow users to disable text prediction/autocomplete/autosuggest without a custom renderer or effect.

## Implementation

Add bindable property to `Entry`:

```csharp
public static readonly BindableProperty IsTextPredictionEnabledProperty = BindableProperty.Create("IsTextPredictionEnabled", typeof(bool), typeof(Entry), true);

```

Renderers will need to be updated accordingly.

### Android

- `IsTextPredictionEnabled == true`: Default. Text prediction is enabled.
- `IsTextPredictionEnabled == false`: Text prediction is disabled.

### iOS

- `IsTextPredictionEnabled == true`: Default. Text prediction is enabled. Note: may also disable autocorrect.
- `IsTextPredictionEnabled == false`: Text prediction is disabled.

### UWP

- `IsTextPredictionEnabled == true`: Default. Text prediction  is enabled.
- `IsTextPredictionEnabled == false`: Text prediction is disabled.

## Implications for CSS

None

## Backward Compatibility

Since we are adding new properties, there should be no compatibility problems as long as:

1. We ensure that an `Entry`'s text prediction defaults to enabled, as is the expected behavior now,
2. We ensure that any Effects or Custom Renderers that may have been created to produce this behavior take precedence over any values that we set, to the best of our ability.

Third party renderers will need to be updated to ensure that this functionality is officially supported.

## Difficulty : Easy-Medium

This is an easy-medium change and great for a first-time contributor. Biggest challenge will be ensuring that disabling text prediction does not also disable other features of the `Entry` view on all platforms.

### PR DESCRIPTION
**Pull request by ntherning**
Makes it easier for users to disable text prediction/autocomplete/autosuggest.
Fixes #1677.

### Description of Change ###

Added the `Entry.IsTextPredictionEnabledProperty` field and `Entry.IsTextPredictionEnabled` property. Modified `EntryRenderer.OnElementPropertyChanged()` for Android, iOS and UWP to react to changes to the new `IsTextPredictionEnabled` property.

On Android setting `IsTextPredictionEnabled` to `false` sets the `InputTypes.TextFlagNoSuggestions` flag on the native control's `InputType`.

On UWP the native control's `IsTextPredictionEnabled` will be set to the value of the `Entry.IsTextPredictionEnabled` property.

On iOS the native control's `AutocorrectionType` property will be set to `No` when `Entry.IsTextPredictionEnabled` is `false`.

### Bugs Fixed ###

- Fixes #1677 [Enhancement] Entry: Control over text-prediction

### API Changes ###

Added:
- `bool Entry.IsTextPredictionEnabled { get; set; } //Bindable Property`

### Behavioral Changes ###

When `IsTextPredictionEnabled ` is set to `false` (default is `true`) and no `CustomKeyboard` is in use the native text prediction feature will be disabled. This means that `CustomKeyboard` always has precedence. If a `Keyboard` has been set which disables text prediction, the `IsTextPredictionEnabled ` property is ignored. The property cannot be used to enable text prediction for a `Keyboard` which explicitly disables it.

**NOTE**: I discovered that Android 7.1 on my Nexus 6 totally ignores the TextFlagNoSuggestions so neither IsSpellCheckEnabled nor IsTextPredictionEnabled have any effect on my device. Googled around and it seems like turning off suggestions is impossible on some devices, probably depending on the keyboard being used. There doesn't seem to be much one can do about this.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Core/Entry.cs [hunk 1]
```diff
@@ -26,6 +26,8 @@ public class Entry : InputView, IFontElement, ITextElement, ITextAlignmentElemen
 
 		public static readonly BindableProperty FontAttributesProperty = FontElement.FontAttributesProperty;
 
+		public static readonly BindableProperty IsTextPredictionEnabledProperty = BindableProperty.Create("IsTextPredictionEnabled", typeof(bool), typeof(Entry), true);
```
Comment (samhouts) [inline Xamarin.Forms.Core/Entry.cs]: can use `nameof(IsTextPredictionEnabled)` here

**Comment by rmarinho** [comment]
looks good @ntherning can you address @samhouts feedback, please add an example of the behaviour on Android. Thanks

**Comment by ntherning** [comment]
@rmarinho Sure, I will update the test as instructed.

File: Xamarin.Forms.Core/Entry.cs (hunk 1 above)
Comment (jassmith) [reply Xamarin.Forms.Core/Entry.cs]: Also set default binding mode to OneTime

**Comment by ntherning** [comment]
@jassmith I tried to set `OneTime` like this

```
public static readonly BindableProperty IsTextPredictionEnabledProperty = BindableProperty.Create(nameof(IsTextPredictionEnabled), typeof(bool), typeof(Entry), true, BindingMode.OneTime);

```

But then this line fails:
```
var entryNoTextPrediction = new Entry { IsTextPredictionEnabled = false };
```
in my `Issue1677.cs` with the exception
```
System.TypeInitializationException
  HResult=0x80131534
  Message=The type initializer for 'Xamarin.Forms.Entry' threw an exception.

Inner Exception 1:
ArgumentException: Not a valid type of BindingMode
```
at
```
Xamarin.Forms.Core.dll!Xamarin.Forms.Entry.IsTextPredictionEnabled.set(bool value) Line 96
	at C:\p\Xamarin.Forms\Xamarin.Forms.Core\Entry.cs(96)
Xamarin.Forms.Controls.dll!Xamarin.Forms.Controls.Issue1677.Init() Line 20
	at C:\p\Xamarin.Forms\Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\Issue1677.cs(20)
Xamarin.Forms.Controls.dll!Xamarin.Forms.Controls.TestContentPage.TestContentPage() Line 349
	at C:\p\Xamarin.Forms\Xamarin.Forms.Controls.Issues\Xamarin.Forms.Controls.Issues.Shared\TestPages\TestPages.cs(349)
Xamarin.Forms.Controls.dll!Xamarin.Forms.Controls.Issue1677.Issue1677()
Xamarin.Forms.Controls.dll!Xamarin.Forms.Controls.App.App() Line 33
	at C:\p\Xamarin.Forms\Xamarin.Forms.Controls\App.cs(33)
Xamarin.Forms.ControlGallery.WindowsUniversal.exe!Xamarin.Forms.ControlGallery.WindowsUniversal.MainPage.MainPage() Line 35
	at C:\p\Xamarin.Forms\Xamarin.Forms.ControlGallery.WindowsUniversal\MainPage.xaml.cs(35)
Xamarin.Forms.ControlGallery.WindowsUniversal.exe!Xamarin.Forms.ControlGallery.WindowsUniversal.Xamarin_Forms_ControlGallery_WindowsUniversal_XamlTypeInfo.XamlTypeInfoProvider.Activate_4_MainPage() Line 255
	at C:\p\Xamarin.Forms\Xamarin.Forms.ControlGallery.WindowsUniversal\obj\x86\Debug\XamlTypeInfo.g.cs(255)
Xamarin.Forms.ControlGallery.WindowsUniversal.exe!Xamarin.Forms.ControlGallery.WindowsUniversal.Xamarin_Forms_ControlGallery_WindowsUniversal_XamlTypeInfo.XamlUserType.ActivateInstance() Line 476
	at C:\p\Xamarin.Forms\Xamarin.Forms.ControlGallery.WindowsUniversal\obj\x86\Debug\XamlTypeInfo.g.cs(476)
[Native to Managed Transition]
[Managed to Native Transition]
Xamarin.Forms.ControlGallery.WindowsUniversal.exe!Xamarin.Forms.ControlGallery.WindowsUniversal.App.OnLaunched(Windows.ApplicationModel.Activation.LaunchActivatedEventArgs e) Line 80
	at C:\p\Xamarin.Forms\Xamarin.Forms.ControlGallery.WindowsUniversal\App.xaml.cs(80)
```

Any idea what I'm doing wrong?

**Comment by jassmith** [comment]
You need to rebase on master :)

**Comment by ntherning** [comment]
Updated test and now using `BindingMode.OneTime` as requested.

**NOTE**: I discovered that Android 7.1 on my Nexus 6 totally ignores the `TextFlagNoSuggestions` so neither `IsSpellCheckEnabled` nor `IsTextPredictionEnabled` have any effect on my device. Googled around and it seems like turning off suggestions is impossible on some devices, probably depending on the keyboard being used. There doesn't seem to be much one can do about this. Works as expected in the emulator though. Will add this to the PR description.

**Pull request merged**